From 1948738b2f6608c790a53a008dd73ebcf85be546 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sun, 9 Dec 2018 22:26:09 +0530
Subject: [PATCH 0853/1051] soc: octeontx2: Fix resource alloc debugfs dump

This patch fixes the RVU block LFs resource allocation
status debugfs dump by skipping PF0.

Change-Id: If5321386da8ec4045b6ef77382b06d179840aabb
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1760
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_debugfs.c | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_debugfs.c b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
index 10e64e82844e..e6e189087831 100644
--- a/drivers/soc/marvell/octeontx2/rvu_debugfs.c
+++ b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
@@ -101,7 +101,7 @@ static ssize_t rvu_dbg_rsrc_attach_status(struct file *filp,
 	int lf, pf, vf, pcifunc;
 	struct rvu_block block;
 	int bytes_not_copied;
-	int buf_size = 1024;
+	int buf_size = 2048;
 	char *buf;
 
 	/* don't allow partial reads */
@@ -123,8 +123,19 @@ static ssize_t rvu_dbg_rsrc_attach_status(struct file *filp,
 	for (pf = 0; pf < rvu->hw->total_pfs; pf++) {
 		for (vf = 0; vf <= rvu->hw->total_vfs; vf++) {
 			pcifunc = pf << 10 | vf;
-			go_back = scnprintf(&buf[off], buf_size - 1 - off,
-					    "0x%3x\t\t", pcifunc);
+			if (!pcifunc)
+				continue;
+
+			if (vf) {
+				go_back = scnprintf(&buf[off],
+						    buf_size - 1 - off,
+						    "PF%d:VF%d\t\t", pf,
+						    vf - 1);
+			} else {
+				go_back = scnprintf(&buf[off],
+						    buf_size - 1 - off,
+						    "PF%d\t\t", pf);
+			}
 
 			off += go_back;
 			for (index = 0; index < BLKTYPE_MAX; index++) {
-- 
2.17.1

